### PR TITLE
Use exasol-toolbox

### PR DESCRIPTION
**Pull request by tomuben**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -36,7 +36,29 @@ build-backend = "poetry.core.masonry.api"
...
 toml = ">=0.10.2"
 configobj = "^5.0.8" # moved to dev dependencie so security alert does not propagate further up
+exasol-toolbox = ">=0.14.0"
 
 
 [tool.poetry.scripts]
 exaslct = 'exasol_script_languages_container_tool.main:main'
+
+[[tool.mypy.overrides]]
+module = ["exasol_integration_test_docker_environment.*", "luigi.*", "docker.*",
```
Comment (tkilias) [inline pyproject.toml:46]: While I see this, we probably add a py.typed file

https://blog.whtsky.me/tech/2021/dont-forget-py.typed-for-your-typed-python-package/

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+jobs:
+
+  lint-job:
+    name: Linting and Type checks (Python-${{ matrix.python-version }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10", "3.11", "3.12"]
```
Comment (kaklakariada) [inline .github/workflows/checks.yml:14]:
```suggestion
        python-version: ["3.12"]
```
Is it necessary to lint with all these python versions?

File: .github/workflows/checks.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run Tests
+        run: poetry run nox -s lint
+
+      - name: Run type-check
+        run: poetry run nox -s type-check
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v3
```
Comment (kaklakariada) [inline .github/workflows/checks.yml:32]:
```suggestion
        uses: actions/upload-artifact@v4
```
You might need to use a unique artifact name if you still use multiple python versions.

**Review by kaklakariada — DISMISSED** [review]

File: pyproject.toml (hunk 1 above)
Comment (tomuben) [reply pyproject.toml:46]: Done

File: .github/workflows/checks.yml (hunk 2 above)
Comment (tomuben) [reply .github/workflows/checks.yml:14]: There might be subtle changes among the different Python versions. I used the `BucketFS-Python` Project as example (see https://github.com/exasol/bucketfs-python/blob/c9e203dcf86c762997a4d67903b0e02091afcc06/.github/workflows/checks.yml#L28) , which I thing created all the github actions from the `exasol-toolbox`. So I guess this is kind of our standard. Maybe @tkilias can add something.

File: .github/workflows/checks.yml (hunk 3 above)
Comment (tomuben) [reply .github/workflows/checks.yml]: Done

**Review by kaklakariada — DISMISSED** [review]

**Review by tkilias — APPROVED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**